server/sonar-web/src/main/js/apps/tutorials/onboarding/ProjectWatcher.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import { getTasksForComponent } from '../../../api/ce';
+import { STATUSES } from '../../../apps/background-tasks/constants';
+import { translate } from '../../../helpers/l10n';
+
+const INTERVAL = 5000;
+const TIMEOUT = 10 * 60 * 1000; // 10 min
+
+type Props = {
+  onFinish: () => void,
+  onTimeout: () => void,
+  projectKey: string
+};
+
+type State = {
+  inQueue: boolean,
+  status: ?string
+};
+
+export default class ProjectWatcher extends React.PureComponent {
+  interval: number;
+  mounted: boolean;
+  props: Props;
+  timeout: number;
+  state: State = {
+    inQueue: false,
+    status: null
+  };
+
+  componentDidMount() {
+    this.mounted = true;
+    this.watch();
+    this.timeout = setTimeout(this.props.onTimeout, TIMEOUT);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+    clearInterval(this.timeout);
+    this.mounted = false;
+  }
+
+  watch = () => (this.interval = setTimeout(this.checkProject, INTERVAL));
+
+  checkProject = () => {
+    const { projectKey } = this.props;
+    getTasksForComponent(projectKey).then(response => {
+      if (response.queue.length > 0) {
+        this.setState({ inQueue: true });
+      }
+
+      if (response.current != null) {
+        const { status } = response.current;
+        this.setState({ status });
+        if (status === STATUSES.SUCCESS) {
+          this.props.onFinish();
+        } else if (status === STATUSES.PENDING || status === STATUSES.IN_PROGRESS) {
+          this.watch();
+        }
+      } else {
+        this.watch();
+      }
+    });
+  };
+
+  render() {
+    const { inQueue, status } = this.state;
+
+    if (status === STATUSES.SUCCESS) {
```

Comment:
I'm not sure it's really used, so remove it or add a small timer before redirecting to show the success message.